Alternate layout

pradeepcep opened the pull request:
Spilltweet initially started with a two-column layout that showed 
the editor and preview side-by-side. This was dropped in favour 
of a full-width layout that made editing easier.

But with the current setup, the width of the editor and the width 
of the output image are not consistent. This means that the 
output tends to have an alignment / layout different from what 
you see in the editor.

This PR is to work on Layout Mk3, which I'm hoping will feel 
more intuitive and lead to a better user experience.

Commit:
Add toolbar modes: simple and advanced

In the simple mode, which is displayed by default, there are only
a few buttons, along with the modeswitch button and preview button.

The advanced or full mode contains all the toolbar buttons.

On clicking the modeswitch button, the toolbar is expanded to show
all the buttons, and the modeswitch button becomes a 'close' button.
If this is clicked, the toolbar reverts to the simple view.

cepradeep commented:
Progress so far:

 - When no text is present:
   ![screenshot from 2017-02-27 15-00-49](https://cloud.githubusercontent.com/assets/6213734/23355527/27a3706a-fcfc-11e6-8efb-a15e76cfe565.png)

 - When text is present:
   ![screenshot from 2017-02-27 15-02-26](https://cloud.githubusercontent.com/assets/6213734/23355580/61359d4e-fcfc-11e6-9946-b7287a48019a.png)

Commit:
Add Live Preview

Added a right sidebar that continually updates according to the
content in the editor. It displays a small preview image, which when
clicked will open the full-size preview.

Removed the Preview button from the toolbar as it is now redundant.

Added a 'Post to Twitter' button on top of the right sidebar. For
simple posts, this should allow the user to finish everything on
a single page, with just one click.

Commit:
Edit info messages accordingly since Preview button is removed

Commit:
Prevent scroll to top on updating Live Preview

On updating the live preview, the document automatically scrolled
to the top, which made editing long documents impossible. This fix
works by storing the scroll position before updating the live
preview and setting the scroll posiiton to the stored value after
the update.

Commit:
Ability to modify some basic properties of the output image

For now Width, Height, Padding and Border-weight can be modified.

TODO: Implement some sort of validation for the image properties
inputs, and add keyboard support (i.e. hitting Enter should update
the properties, as opposed to the current setup that requires the
user to click the 'Apply' button).

cepradeep commented:
Progress:

![screenshot from 2017-02-28 00-30-23](https://cloud.githubusercontent.com/assets/6213734/23374779/c6c4d752-fd4b-11e6-849d-c408df17ad5a.png)

Commit:
Show current values of output image properties alongside inputs

The values will be dynamically updated along with the Live Preview.
When the user sets a value for a property, it will be remembered,
and that property will no longer be automaticlly updated. A property
can be reset to the default value by leaving it blank.
